sc-app-deploy/sc-app-deploy.py

```diff
@@ -18,75 +19,100 @@
+                                help='script is called by automatic scheduler')
+
+    branch_base_names = ('develop', 'master', 'release', 'hotfix', 'feature')
+    subp = parser.add_subparsers(title='Branches', metavar='\n  '.join(branch_base_names))
+
+    # develop PARSER
+    develop_parser = subp.add_parser('develop',
+                                      usage=(' develop '),
+                                      description='Deploy latest images from develop branch')
+    add_standard_args(develop_parser,
+                      ('scheduled', 'team_number'))
+    develop_parser.set_defaults(func=deployImages)
+
+    # master PARSER
+    master_parser = subp.add_parser('master',
+                                      usage=(' master '),
+                                      description='Deploy latest images from master branch')
+    add_standard_args(master_parser,
+                      ('team_number', 'ticket_id'))
+    master_parser.set_defaults(func=deployImages)
+
+    # release PARSER
+    release_parser = subp.add_parser('release',
+                                      usage=(' release '),
+                                      description='Deploy latest images from release branch')
+    add_standard_args(release_parser,
+                      ('team_number', 'ticket_id'))
+    release_parser.set_defaults(func=deployImages)
+    # hotfix PARSER
+    hotfix_parser = subp.add_parser('hotfix',
+                                      usage=(' hotfix '),
+                                      description='Deploy latest images from hotfix branch of team')
+    add_standard_args(hotfix_parser,
+                      ('team_number', 'ticket_id'))
+    hotfix_parser.set_defaults(func=deployImages)
+
+    # feature PARSER
+    feature_parser = subp.add_parser('feature',
+                                      usage=(' feature '),
+                                      description='Deploy latest images from feature branch of team')
+    add_standard_args(feature_parser,
+                      ('team_number', 'ticket_id'))
+    feature_parser.set_defaults(func=deployImages)
+
+    args = parser.parse_args()
+    return args
+    
 
 if __name__ == '__main__':
     try:
+        if sys.version_info[0] < 3 or sys.version_info[1] < 6:
+            print("This script requires Python version 3.6")
+            print("Python version")
+            print (sys.version)
+            print("Version info.")
+            print (sys.version_info)          
+            print(os.environ['PATH'])
+            sys.exit(1)
+
         initLogging()
+        parsedArgs = parseArguments()
         logging.info('Call arguments given: %s' % sys.argv[1:])
-
-        decryptedSshKeyFile = None
-        if sad_secrets.secret_helper.isPassphraseSet():
-            decryptedSshKeyFile="travisssh"
-            sad_secrets.secret_helper.gpgDecrypt(decryptedSshKeyFile)
+        if hasattr(parsedArgs, 'func'):
+            parsedArgs.func(sys.argv[1], parsedArgs)
```

Comment:
To me this looks very cryptic. Readability is not good and debugging will be difficult. Should be doable without the relatively complicated subparser. As a result the help printout is currently not really helpful and actually completely false. I get just the possible branch names printed out without any parameters or help texts.

Reply:
I have changed the usage, so it should be correct. DUe to the different parameters I see subparsers as a useful option.

Reply:
Usage help is OK now with the exception of the --scheduled parameter, where one needs to add dummy parameter. This should be fixed (see comment in workflow script).

Reply:
Parameters changed